### DISCUSSION TITLE
original pageContext lost on error

### DISCUSSION THREAD

**Discussion by Ultre00** [vikejs/vike#1110] [Discussion]
Currently, using version `0.4.133`

Not sure if this is a bug, or if this is intended. But when a server error occurs, I am rendering a custom error page. I am using the `_error.page.server.ts` for this. This server file is hit on error, but I am missing data that is passed in the default `pageContext`.

Context: In the `_default.page.route.ts` I am extracting the locale as provided in one of the examples. So the `pageContext` has a `locale` prop. When debugging the `pageContext` that is passed in the `onBeforeRender` of the `_error.page.server.ts` is missing the locale. Therefore, I am unable to show a localized error message. Is there something I am overlooking, or is this currently not possible?

How can I access the original `pageContext` in the `_error.page.ts` ?

**Comment by brillout** [comment]
> How can I access the original `pageContext` in the `_error.page.ts` ?

Have you seen https://vite-plugin-ssr.com/usePageContext?

**Reply by Ultre00** [reply]
Yes, I am using the custom usePageContext hook. The thing is that there is definitely a pageContext, it is just a new instance in the _error.page. So my original pageContext data is lost.

**Reply by brillout** [reply]
https://github.com/brillout/vite-plugin-ssr/issues/1112

How urgent is it? I'm fairly busy with high prio tickets, but I understand this can be a blocker for you. Let me know.

(Also in case your company is up for it: https://github.com/sponsors/brillout.)

**Reply by Ultre00** [reply]
Thanks, I will follow #1112. It is not a high prio for now. I duplicated my route logic in the `_error.page.server.ts` for now, so I can move forward.
